src/algokit/core/deploy.py

```diff
@@ -0,0 +1,118 @@
+# 1. User can call algokit deploy to different networks
+# 2. By default algokit cli contains configs for testnet and mainnet
+# 3. User can overwrite them by creating a config file in the project root
+
+
+import logging
+import os
+import sys
+from pathlib import Path
+
+import click
+import httpx
+from dotenv import dotenv_values
+
+from algokit.core import proc
+from algokit.core.conf import get_algokit_config
+from algokit.core.constants import ALGOKIT_CONFIG, ALGORAND_NETWORKS, AlgorandNetworkConfiguration
+
+if sys.version_info >= (3, 11):
+    pass
+else:
+    pass
+
+logger = logging.getLogger(__name__)
+
+
+def get_genesis_network_name(deploy_config: dict) -> str | None:
+    """
+    Get the network name from the genesis block.
+    :param deploy_config: Deploy configuration.
+    :return: Network name.
+    """
+
+    algod_server = deploy_config.get("ALGOD_SERVER")
+
+    if not algod_server:
+        raise click.ClickException("Missing ALGOD_SERVER in deploy configuration.")
+
+    try:
+        genesis_response = httpx.get(f"{algod_server}/genesis")
+        genesis_response.raise_for_status()
+        return genesis_response.json()["network"] if genesis_response.status_code == 200 else None  # noqa: PLR2004
+    except httpx.HTTPError:
+        logger.warning(f"Failed to load network name from {algod_server} due to HTTP error.", exc_info=True)
+        return None
+
+
+def load_deploy_config(network: str, project_dir: Path) -> dict:
+    """
+    Load the deploy configuration for the given network.
+    :param network: Network name.
+    :param project_dir: Project directory path.
+    :return: Deploy configuration.
+    """
+
+    env_file_path = project_dir / f".env.{network}"
```

Comment:
we should be loading from `.env` also if it exists - `.env.{foo}` should take precedence though.
it might be easier here to:
1) copy `os.environ`
2) if network is a known name, update the dict with the defaults
3) load in from .env
4) load in from .env.foo if it it exists. if it doesn't exist and foo isn't a known network name, maybe that's an error?

Reply:
also, this will suffer from the same problem of requiring the user to be in the project root to run deploy, maybe it should walk up to discover env file(s) (and maybe also this should be done for algokit.toml - not sure what to do if that doesn't exist though)

Reply:
I wanted to avoid implicit folder scanning as that can get a bit tricky since we always require the toml and env files to be placed on the same level, we also agreed on this convention with Alessandro Rob and Daniel so far but i agree that it needs some more flexibility to allow for deploying from custom paths. So:

1. .env files are now loaded as well IFF there is no env.network name prefixed files present 
2. if user needs deployment from other folders he can simply supply custom path via `--project-dir`

Reply:
Please resolve if you think this is a good alternative to walking up the dirs otherwise we can continue the thread @achidlow